tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/mqtt/imported/MQTTQueueCleanTest.java

```diff
@@ -0,0 +1,109 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.activemq.artemis.tests.integration.mqtt.imported;
+
+import org.apache.activemq.artemis.api.core.SimpleString;
+import org.apache.activemq.artemis.core.postoffice.Binding;
+import org.apache.activemq.artemis.core.postoffice.PostOffice;
+import org.apache.activemq.artemis.core.server.ActiveMQServer;
+import org.apache.activemq.artemis.core.server.ActiveMQServerLogger;
+import org.apache.activemq.artemis.core.server.Queue;
+import org.junit.Test;
+
+import java.util.HashSet;
+import java.util.Random;
+import java.util.Set;
+
+public class MQTTQueueCleanTest extends MQTTTestSupport {
+
+   private static final ActiveMQServerLogger log = ActiveMQServerLogger.LOGGER;
+
+   @Test
+   public void testQueueCleanWhenConnectionSynExeConnectAndDisconnect() throws Exception {
+      Random random = new Random();
+      Set<MQTTClientProvider> clientProviders = new HashSet<>(11);
+      int repeatCount = 0;
+      String address = "clean/test";
+      String clientId = "sameClientId";
+      String queueName = "::sameClientId.clean.test";
+      //The abnormal scene does not necessarily occur, repeating 100 times to ensure the recurrence of the abnormality
+      while (repeatCount < 100) {
+         repeatCount++;
+         int subConnectionCount = random.nextInt(50) + 1;
+         int sC = 0;
+         try {
+            //Reconnect at least twice to reproduce the problem
+            while (sC < subConnectionCount) {
+               sC++;
+               MQTTClientProvider clientProvider = getMQTTClientProvider();
+               clientProvider.setClientId(clientId);
+               initializeConnection(clientProvider);
+               clientProviders.add(clientProvider);
+               clientProvider.subscribe(address, AT_LEAST_ONCE);
+            }
+         } finally {
+            for (MQTTClientProvider clientProvider : clientProviders) {
+               clientProvider.disconnect();
+            }
+            clientProviders.clear();
+            assertTrue(waitForBindings(server, queueName, false, 0, 0, 10000));
```

Comment:
From what I can tell the overridden version of `waitForBindings` isn't necessary. You could just use something like:

`assertTrue(Wait.waitFor(() -> server.locateQueue(SimpleString.toSimpleString(queueName)) == null));`

Reply:
I pushed an update to addressyour suggestion. And removed `synchronized `from some methods ( `MQTTConnectionManager.connect`, `MQTTConnectionManager.disconnect`, `MQTTSubscriptionManager.removeSubscription` ).